pandas/core/frame.py

```diff
@@ -3484,7 +3484,9 @@ def _setitem_array(self, key, value):
                 for k1, k2 in zip(key, value.columns):
                     self[k1] = value[k2]
             else:
-                indexer = self.loc._convert_to_indexer(key, axis=1)
+                indexer = self.loc._get_listlike_indexer(
```

Comment:
```self.columns.get_indexer(key)```?

Reply:
Looks like this returns something different, will have to keep digging